docs/development/creating-your-first-event-listener.md

```diff
@@ -0,0 +1,81 @@
+---
+id: creating-your-first-event-listener
+title: Creating Your First Event Listener
+sidebar_label: Creating Your First Event Listener
+---
+
+## Prerequisites
+
+This guide assumes that you have a [plugin environment](creating-a-first-plugin.md) setup.
+If you have not already, follow that guide first.
+
+## Event Concept
+
+Many actions done by players, blocks or mobs trigger an event. Technically: a class extending `Event`. Events allow you to know, edit or cancel default behaviour.
+For your plugin to get informed when an event happens, you need to create a class implementing `EventListener` (a marker interface):
+
+```java
+public class MyFirstEventListener implements EventListener {}
+```
+
+Now we marked the class for us and the api as event listener. Then we have to register our class to the API. We do that for example in the `onInstall` method in our `TestPlugin` class. Note that we use the method [`registerActiveWorldsListener`](https://janmm14.de/static/gomint/gomint.api/io/gomint/plugin/Plugin.html#registerActiveWorldsListener(io.gomint.event.EventListener)) which automatically filters events so we only get events taking place in any of the [plugin's active worlds](../get-started/plugin-world-restriction.md).
+
+```java
+  @Override
+  public void onInstall() {
+    registerActiveWorldsListener(new MyFirstEventListener());
+  }
+```
+Next up we need to define which event(s) we want to listen to in this event listener.
+This is done simply by creating a method annotated with `@EventHandler` which takes one argument (the event we want to listen to) and returns `void`:
+
+```java
+public class MyFirstEventListener implements EventListener {
+  @EventHandler
+  public void onExplode(EntityExplodeEvent event) {
+    // this method gets called every time an entity explodes
+  }
+}
+```
+
+Now we just need to implement some logic. We can praise explosions for example with a chat message sent to every player:
+
+
+```java
+public class MyFirstEventListener implements EventListener {
+  @EventHandler
+  public void onExplode(EntityExplodeEvent event) {
+    GoMint.instance().onlinePlayers().forEach(p -> p.sendMessage("Hooray, " + event.affectedBlocks().size() + " are gone!"));
```

Comment:
Then I guess I need to remove some world thread checks again, because we made packet queueing thread safe. Will also edit the example to restrict it to plugin's worlds.

Reply:
sendMessage is thread safe